scripts/check-upgrade-guide-exists.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+: "${1?"Tag of new release must be provided"}"
+
+RELEASE_TAG=$1
+RELEASE_NUMBER=$(echo "${RELEASE_TAG}" | tr -d v)
+
+IFS='.' read -r MAJOR MINOR PATCH <<< "$RELEASE_NUMBER"
+
+# Check if it's a major release (patch version is 0)
+if [ "$PATCH" -eq 0 ]; then
+    UPGRADE_GUIDE_PATH="website/docs/guides/$MAJOR.$MINOR.$PATCH-upgrade-guide.html.markdown"
+    echo "Checking for the presence of $UPGRADE_GUIDE_PATH"
+    if [ ! -f "$UPGRADE_GUIDE_PATH" ]; then
+        echo "Stopping release process, upgrade guide $UPGRADE_GUIDE_PATH does not exist. Please visit our releasing documentation for more details."
```

Comment:
should we add the link or do you think it's not necessary?
https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/RELEASING.md#verify-upgrade-guide-is-defined

Reply:
Preferred to keep it more generic so the reference does not breaking moving forward, but no strong opinion

Reply:
we could add a link to the RELEASE.md instead of the exact section